src/crimson/osd/osd.cc

```diff
@@ -391,12 +395,70 @@ seastar::future<> OSD::_send_alive()
   }
 }
 
+/*
+  The OSD's Admin Socket object created here has three servers (i.e. - blocks of commands
+  to handle) registered to it:
+  - OSD's specific commands are handled by the OSD object;
+  - CephContext general commands are registered by the OSD's CephContext, and
+  - there are some common commands registered to be directly handled by the AdminSocket object
+    itself.
+
+  A note regarding the CephContext pointer passed to AdminSocket at creation: that one is used
+  by AdminSocket internally (mostly related to error reporting by legacy code), and is unrelated to
+  the role played by the (same) CephContext object as a command handler.
+*/
+seastar::future<> OSD::start_asok_admin()
+{
+  asok = seastar::make_lw_shared<crimson::admin::AdminSocket>();
+  auto asok_path = local_conf().get_val<std::string>("admin_socket");
+
+  return asok->start(asok_path).then([this] {
+    // register OSD-specific admin-socket hooks
+    admin = std::make_unique<crimson::admin::OsdAdmin>(this);
```

Comment:
also, i'd suggest construct `admin` in the constructor of `OSD`.

Reply:
done

Reply:
OK